scripts/irods/test/test_pam_password.py

```diff
@@ -0,0 +1,225 @@
+
+            IrodsController().restart()
+
+
+class test_iadmin_set_pam_password_config(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.admin = session.mkuser_and_return_session('rodsadmin', 'otherrods', 'rods', lib.get_hostname())
+
+        self.default_setup = {
+            "no_extend": "",
+            "password_length": "",
+            "password_max_time": "",
+            "password_min_time": ""
+        }
+
+        _, out, _ = self.admin.assert_icommand(['iadmin', 'get_pam_password_config'], 'STDOUT', 'password_length')
+        self.original_configs = json.loads(out)
+
+        # Set the pam_password configuration values to the defaults.
+        for option_name in self.default_setup:
+            self.admin.assert_icommand(
+                ['iadmin', 'set_pam_password_config', option_name, self.default_setup[option_name]])
+
+    @classmethod
+    def tearDownClass(self):
+        # Put everything back to the original configuration settings.
+        for option_name in self.original_configs:
+            self.admin.assert_icommand(
+                ['iadmin', 'set_pam_password_config', option_name, self.original_configs[option_name]])
+
+        with session.make_session_for_existing_admin() as admin_session:
+            self.admin.__exit__()
+            admin_session.assert_icommand(['iadmin', 'rmuser', self.admin.username])
+
+    def test_no_parameters(self):
+        self.admin.assert_icommand(
+            ['iadmin', 'set_pam_password_config'], 'STDERR', 'Invalid pam_password configuration option')
+
+        # Assert that nothing changed due to the error.
+        expected_output = self.default_setup
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', json.dumps(expected_output, indent=4))
+
+    def test_invalid_option_name(self):
+        self.admin.assert_icommand(
+            ['iadmin', 'set_pam_password_config', 'invalid'], 'STDERR', 'Invalid pam_password configuration option')
+
+        # Assert that nothing changed due to the error.
+        expected_output = self.default_setup
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', json.dumps(expected_output, indent=4))
+
+    def test_empty_value(self):
+        option_name = 'password_length'
+        option_value = 500
+        self.admin.assert_icommand(['iadmin', 'set_pam_password_config', option_name, str(option_value)])
+
+        # Assert that the value has been updated to the expected new value.
+        expected_output = self.default_setup.copy()
+        expected_output['password_length'] = str(option_value)
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', f'"{option_name}": "{option_value}"')
+
+        # Assert that the value has been set to an empty string.
+        self.admin.assert_icommand(['iadmin', 'set_pam_password_config', option_name])
```

Comment:
Oh, is NOT passing an empty string a feature of this subcommand?

Reply:
Not passing an empty string will actually set the value of the configuration option, like this:
```bash
iadmin set_pam_password_config password_max_length 500
```

Setting the value to nothing by passing an empty string will instruct the system to use the default value, like this:
```bash
iadmin set_pam_password_config password_max_length
```

Happy to try something else if it's more intuitive

Reply:
It feels a little surprising to not have to provide at least `''` as the empty string.

I'd imagine `argc` in `main` would match 3 in that case. How does iadmin not crash?

Reply:
The `cmdToken` array elements are initialized to empty strings here: https://github.com/irods/irods_client_icommands/blob/9d06556c3f0a9c2424bcd07d93889af02d949f2e/src/iadmin.cpp#L1694-L1698

Reply:
Ehhh sorta. I think that implicitly this is supported in the sense that an arg can be both required and empty, but I think most subcommands require a non-empty string. Either way, I'm not particularly attached to this approach, so we can do something else here.

How should the admin indicate that the default value should be used? A special string? A negative value?

Reply:
I think "default" seems like a good option. 0 is a valid value for one of these configurations and hyphen is confusing (to me). Thoughts, @trel?

Reply:
`default` seems fine. There's certainly no room for confusion there.

Will await @trel's thoughts.

Reply:
feels like empty string is problematic / different from others - and causes more edge cases?

can we just require a value?  and if they want the default, then ... we can document the magic string(s) 'default' and/or 'reset'?     hmmm....

Reply:
really don't like hyphen or 0 or negative.

Reply:
Still need to look at this. I think we decided that "default" can be sent via `iadmin` which will cause the default value to be stored in `R_GRID_CONFIGURATION`.

Reply:
Created https://github.com/irods/irods/issues/7274 for the `R_GRID_CONFIGURATION` changes. That will happen in a different PR.